File: dev-docs/snippets/cache/key.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Build a deterministic cache key from a base string and optional params.
export function key(client: JsBaoClient) {
  // #region example
  const cacheKey = client.cache.key("posts", { page: 1, sort: "recent" });
  // #endregion example
  return cacheKey;
}

File: dev-docs/snippets/cache/key.swift
import JsBaoClient

// Build a deterministic cache key from a base string and optional params.
func key(client: JsBaoClient) {
  // #region example
  let cacheKey = client.cache.key("posts", params: ["page": 1, "sort": "recent"])
  // #endregion example
  _ = cacheKey
}
